index.js

```diff
@@ -45,7 +46,7 @@ module.exports = async (target, options) => {
 	// double-quotes through the “double-quotes on Windows caveat”, but it
 	// can be used on any platform.
 	if (options.url) {
-		target = encodeURI(target);
+		target = new url.URL(target).href;
```

Comment:
You can just use `URL` as a global.

Reply:
The global version of `URL` has been available since Node [v10.0.0](https://nodejs.org/api/url.html#url_class_url). I wasn’t quite sure sure which versions of Node `open` supports, so I went with `url.URL` since it’s been available since Node v7.0.0, v6.13.0.

Reply:
Oh my fault. I forgot `open` still targets Node.js 8.